### ISSUE TITLE
Autocomplete suggestions are not correct when having multiple windows with different directories inside one iTerm tab

### ISSUE THREAD

**Issue by reck753** [withfig/fig#469]
### Sanity checks

- [X] I have searched [github.com/withfig/fig/issues](https://github.com/withfig/fig/issues?q=) and there are no duplicates of my issue

### Description:
The issue seems to happen when having multiple windows opened in one iTerm tab with different directories. To whichever directory I navigate last, its autocompletes are available to the rest of the windows regardless of the directory.
For example, I have a monorepo setup and I open three iTerm windows within one tab. First, in two of them, I navigate to the root of the project in which I have a specific `package.json`, with specific custom `yarn` commands. Those commands will appear in the autocomplete suggestion as they should. Now, in the third window, I navigate to one subdirectory (one of the packages of the monorepo), which has its own `package.json` with its own specific custom commands. In that third window, again, correct commands appear for autocomplete, but however, for the rest of the windows (that are set for the root directory of the project) they are not correct and autocomplete suggests commands of the subdirectory.

https://user-images.githubusercontent.com/58557161/134296120-5c2904f8-b2df-4444-8b66-284e38e231d4.mov

This is the information I provided to Wes from `Fig Team`, so if you need any additional information just let me know.

### Environment

```bash
Version 1.0.49 (B269) [U.S.]
UserShell: /bin/zsh
Bundle path: /Applications/Fig.app
Autocomplete: true
Settings.json: true
CLI installed: true
CLI tool path: /Users/aleksa/.fig/bin/fig
Accessibility: true
Number of specs: 206
SSH Integration: false
Tmux Integration: false
Keybindings path: <none>
iTerm Integration: true
Hyper Integration: false
VSCode Integration: true
Docker Integration: false
Symlinked dotfiles: false
Only insert on tab: false
Installation Script: true
PseudoTerminal Path: /usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin
SecureKeyboardInput: false
SecureKeyboardProcess: <none>
Current active process: /bin/zsh (94741) - ttys017
Current working directory: /Users/aleksa
Current window identifier: 62518/% (com.googlecode.iterm2)
PATH: /usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/Users/aleksa/.fig/bin
FIG_INTEGRATION_VERSION: 4
```

**Comment by mschrage** [comment]
@reck753 it looks like your iTerm integration isn’t working correctly! What version of iTerm are you on?

**Comment by reck753** [comment]
@mschrage Hi Matt, the version of iTerm at the time of reporting the issue was 3.3.12. Not long after reporting, I had in mind that the issue could be in the version, since I remembered I didn't update it in ages. 
After upgrading to the latest version (3.4.10), it seems it got fixed. I didn't want to leave an update here since after some time one of the windows got messed up (it started showing wrong autocompletes), but after restarting the iTerm it went back to normal.
I will happily continue testing it and if the random issue happens again, I will try to catch a pattern in which it happens (if there is one) and I will report it back at you!
